File: app/src/main/java/ga/forntoh/bableschool/model/Video.java
package ga.forntoh.bableschool.model;

public class Video {

    private String title, author, duration, url, thumbnail;

    Video(String title, String author, String duration, String url, String thumbnail) {
        this.title = title;
        this.author = author;
        this.duration = duration;
        this.url = url;
        this.thumbnail = thumbnail;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }

    public String getDuration() {
        return duration;
    }

    public String getUrl() {
        return url;
    }
}

File: app/src/main/java/ga/forntoh/bableschool/data/model/main/Video.kt
package ga.forntoh.bableschool.data.model.main

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import ga.forntoh.bableschool.data.model.groupie.ItemVideo

@Entity(
        foreignKeys = [ForeignKey(
                entity = Course::class,
                parentColumns = ["code"],
                childColumns = ["courseCode"]
        )]
)
data class Video(
        @ColumnInfo(index = true) var courseCode: String? = null,
        @ColumnInfo var title: String? = null,
        @ColumnInfo var author: String? = null,
        @ColumnInfo var duration: String? = null,
        @PrimaryKey var url: String = "",
        @ColumnInfo var thumbnail: String? = null
)

fun Video.toVideoView() = ItemVideo(courseCode, title, author, duration, url, thumbnail)
